### ISSUE TITLE
Baldur's Gate 1 performance hit in rivers

### ISSUE THREAD

**Issue by JNx666** [narzoul/DDrawCompat#367]
Hi.

I'm using Baldur's Gate 1 v1.3.5521 original from "Dungeons and Dragons Anthology" by Atari, with no modification or fixes.

When I enter Nashkel from the north forest, I have a performance hit, slowing down the game completely. I tried different options from DDrawCompat.ini without good result. I need to pause the game to runs normally, then select where I need to go, and then unpause and wait for the characters to reach their destination. The game goes back to normal speed if it doesn't render any river.

I also tried the Nvidia settings, using integrated graphics or NVIDIA processor. No fix.

I enable the debug options, but the file size is +400mb. DDrawCompat-BGMain2.log: <a href="https://pastebin.com/WQzREGUA">https://pastebin.com/WQzREGUA</a> (Ignore the name folder, I'm not using the EE version)

From here:
<img src="https://i.imgur.com/xnGT8ye.jpeg">
<img src="https://i.imgur.com/A8vgpli.jpeg">

To Nashkel
<img src="https://i.imgur.com/4trSk8U.jpeg">

The same situation when the game tries to render a river here:
<img src="https://i.imgur.com/rAwf96G.png">
<img src="https://i.imgur.com/9UNikuO.jpeg">

I notice the same situation in green river in Fallout 1 v1.0 original, no modification, I don't have a screenshot at the moment, here is the location: https://youtu.be/ymQdconHjnA?t=1196 

Thank you

**Comment by narzoul** [comment]
This is strange. I haven't seen this topic until now, nor have I received any of the usual notification emails about any of your posts in it. I double checked and I don't have notifications disabled for this issue either...

Anyway, is this still an issue with v0.6.0? If yes, I'll try to look into it later.

**Comment by narzoul** [comment]
Hi, I looked at your video and logs anyway, and it looks like the typical thread starvation issue on Windows 11 24H2. v0.6.0 should solve this now, but you can also try the CpuAffinity=all setting, if not. Please reopen this if neither of this helps.

**Comment by narzoul** [comment]
> Do you have another way to communicate? Maybe an email, Telegram or Discord group?

No, I intentionally keep it limited to GitHub. But at least this time I got the notification,
